tests/src/Context/DrupalContext.php

```diff
@@ -97,4 +104,44 @@ public function resetNodes() {
     unset($this->maxNodeId);
   }
 
+  /**
+   * Remember the list of node type.
+   *
+   * @BeforeScenario @reset-node-types
+   */
+  public function rememberCurrentNodeTypes() {
+    $results = db_select('node_type')
+      ->fields('node_type', array('type'))
+      ->execute();
+    if ($results) {
+      foreach ($results as $result) {
+        $this->nodeTypes[] = $result->type;
+      }
+    }
+  }
+
+  /**
+   * Removes any node types created after the last list node type remembered.
+   *
+   * @AfterScenario @reset-node-types
+   */
+  public function resetNodeTypes() {
+    if (!isset($this->nodeTypes)) {
+      return;
+    }
+
+    $results = db_select('node_type')
```

Comment:
Similar like in the previous comment. Usage of the `node_type_get_names()` function would make it more consistent especially that below there is the `node_type_delete()` which is coming from the same functions family. In future that approach would be safer because it relies on the API rather than on the database queries.